### PR TITLE
Allow a setting of 'auto' for fuzziness.

### PR DESCRIPTION
**Pull request by m4olivei**
There are some Lucene parsers out there (in my case working with Elasticsearch), that don't require you to specify a distance on the fuzziness modifier (`~`).

This change allows you to set a fuzziness of `auto`, which will leave any numeric distance off the end of the fuzziness modifier.

### CODE REVIEW

File: src/AbstractFuzzyQuery.php
Comment (pounard) [inline src/AbstractFuzzyQuery.php]: Please use `!==` instead of `!=`

File: src/AbstractFuzzyQuery.php
Comment (pounard) [inline src/AbstractFuzzyQuery.php]: Since test upper uses `is_numeric()` you should use `is_numeric()` here too, or even better test against `self::FUZZY_AUTO` and completely drop `is_numeric()` call.

**Review by pounard — CHANGES_REQUESTED** [review]
Nice to see someone using this API. Patch seems good to me, please just fix the 2 minor issues I raised and it'll be good for merge.

**Comment by m4olivei** [comment]
Thanks for the feedback! All fixed and ready.

Yeah this is totally handy. We're considering using it in the Elasticsearch module for Drupal 8. In case your interested: 

Issue: https://www.drupal.org/node/2910630
WIP PR: https://github.com/nodespark/elasticsearch_connector/pull/19

File: src/AbstractFuzzyQuery.php
Comment (pounard) [inline src/AbstractFuzzyQuery.php]: > We're considering using it in the Elasticsearch module for Drupal 8.

I guess that's why you indented with 2 space instead of 4 :) Just fix those 2 lines and I'll merge right after.

Many thanks!

**Review by pounard — CHANGES_REQUESTED** [review]

File: src/AbstractFuzzyQuery.php
Comment (m4olivei) [reply src/AbstractFuzzyQuery.php]: Ha! Totally. Coming right up.

**Comment by m4olivei** [comment]
@pounard what do you think? Good to merge now? Would love to be able to stop using my fork, to help encourage using this.

**Comment by pounard** [comment]
@m4olivei Yes, it's great, I'm merging it, thanks!

**Comment by pounard** [comment]
I also released the 1.0.2 version, have fun with it!

**Comment by m4olivei** [comment]
Thanks!

**Comment by pounard** [comment]
Sorry I took my time, I'm in vacations :) Don't hesitate to open new issues or pull requests.

**Comment by m4olivei** [comment]
No worries. Enjoy your vacation friend!

**Pull request merged**
